libvast/src/system/indexer_downstream_manager.cpp

```diff
@@ -56,6 +56,27 @@ size_t indexer_downstream_manager::buffered(caf::stream_slot slot) const
   return 0u;
 }
 
+void indexer_downstream_manager::close() {
+  VAST_DEBUG(self(), "closing downstream manager");
+  if (closing)
+    return;
+  for (auto it = partitions.begin(); it != partitions.end();) {
+    if (buffered(**it) == 0u) {
```

Comment:
Can this function get a few comments? The double de-referencing scares me.

Reply:
I added a few interleaved lines, but I did not address the double referencing directly. It's just what you get if you have pointers in a container and iterate over that.

Reply:
I just wanted to know whether it is a safe assumption that `*it != nullptr` at all times. This could also be expressed as an assertion above this line.

Reply:
Assertions are a great idea. Integrated.